taskvine/src/manager/vine_manager_get.c

```diff
@@ -426,12 +426,12 @@ vine_result_code_t vine_manager_get_output_file(struct vine_manager *q, struct v
 	// If the transfer was successful, make a record of it in the cache.
 	if (result == VINE_SUCCESS && (f->cache_level > VINE_CACHE_LEVEL_TASK)) {
 		struct stat local_info;
-		if (stat(f->source, &local_info) == 0) {
+		if (f->type == VINE_BUFFER || stat(f->source, &local_info) == 0) {
 			struct vine_file_replica *replica = vine_file_replica_create(
 					f->type, f->cache_level, local_info.st_size, local_info.st_mtime);
```

Comment:
`local_info.st_size` will be uninitialized here in the case of a buffer.

However, `f->size` was already set in `vine_manager_get_buffer` and I'm sure the information is also already available in `vine_manager_get_file`.  Maybe the `stat` here is unnecessary if `f->size` has been set in all prior cases?

Reply:
Good catch. Let met see about f->size.

Reply:
Actually the st._size would be wrong for directories too.